src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -760,5 +760,22 @@ internal BoundStatement WrapWithVariablesIfAny(CSharpSyntaxNode scopeDesignator,
                                   ImmutableArray.Create(statement))
                         { WasCompilerGenerated = true };
         }
+
+        /// <summary>
+        /// Should only be used with scopes that could declare local functions.
+        /// </summary>
+        internal BoundStatement WrapWithVariablesAndLocalFunctionsIfAny(CSharpSyntaxNode scopeDesignator, BoundStatement statement)
+        {
+            var locals = this.GetDeclaredLocalsForScope(scopeDesignator);
+            var localFunctions = this.GetDeclaredLocalFunctionsForScope(scopeDesignator);
+            if (locals.IsEmpty && localFunctions.IsEmpty)
+            {
+                return statement;
+            }
+
+            return new BoundBlock(statement.Syntax, locals, localFunctions,
+                                  ImmutableArray.Create(statement))
+            { WasCompilerGenerated = true };
+        }
```

Comment:
To avoid the problem of documenting and knowing whether to call `WrapWithVariablesIfAny` or the new method, maybe the new logic could be added to the existing method instead. 
The caller would always use `WrapWithVariablesIfAny` and that existing method would be smart to check if the scope could declare local functions.